packages/odata2ts/src/generator/ModelGenerator.ts

```diff
@@ -12,6 +12,18 @@ export const generateModels: EntityBasedGeneratorFunction = (dataModel, sourceFi
   return generator.generate();
 };
 
+export function reorderModelsByInheritance<T extends ComplexType>(models: T[]): T[] {
+  // bring models in order of inheritance (base classes first)
+  const inheritedModels = models.filter((m) => m.baseClasses.length);
+  const baseModels = models.filter((m) => !m.baseClasses.length);
+  for (const model of inheritedModels) {
+    let index = baseModels.findIndex((m) => m.name == model.baseClasses[0]);
+    index = index == -1 ? baseModels.length : index + 1;
```

Comment:
while this works like a charm for the presented cases, you don't do any ordering for `inheritedModels` which extend models which in turn extend something.

For example, you could change the test and add this as first line:
```
    odataBuilder.addEntityType("GrandChild", "Child", (builder) => builder);
    ...
```
Then again, the order won't be correct, because `GrandChild` gets inserted before `Child` which is wrong again.

Reply:
Moved everything to data model digestion. Now does proper recursive topological sorting.